packages/chevrotain/src/parse/cst/cst_visitor.ts

```diff
@@ -9,8 +9,9 @@ import isUndefined from "lodash/isUndefined"
 import includes from "lodash/includes"
 import { defineNameProp } from "../../lang/lang_extensions"
 import { CstNode, ICstVisitor } from "@chevrotain/types"
+import flatMap from "lodash/flatMap"
 
-export function defaultVisit<IN, OUT>(ctx: any, param: IN): OUT {
+export function defaultVisit<IN, OUT>(ctx: any, param: IN): void {
```

Comment:
good catch 👍 
should we also remove `OUT` from the generic arguments?

Reply:
Yeah, they don't really make sense here. Is there any point to exporting this function?

Reply:
I don't really recall...

The CST Visitor APIs need re-work
#1039

Reply:
Gonna leave it as is for now. https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unnecessary-type-arguments.md could be added in a separate PR.

Reply:
I changed it as if we are going to refactor a none optimal function in the productive code, we might as well refactor it to the most optimal style we know.